gengen/README.md

```diff
@@ -0,0 +1,64 @@
+# gengen
+
+gengen is a filecoin tool that builds the genesis.car, which is used to seed the
+genesis block.
+
+### Example
+
+```
+go-filecon $ ./gengen/gengen --keypath fixtures --out-car fixtures/genesis.car --out-json fixtures/gen.json --config fixtures/setup.json
+```
+
+### Building
+
+The gengen tool expects that you can already build `go-filecoin`. Please refer
+to the README in the root of this project for details.
+
+```
+go build -o gengen main.go
+```
+
+### Usage
+
+```
+Usage of ./gengen:
+  -config string
+    	reads configuration from this json file, instead of stdin
+  -json
+    	sets output to be json
+  -keypath string
+    	sets location to write key files to (default ".")
+  -out-car string
+    	writes the generated car file to the give path, instead of stdout
+  -out-json string
+    	enables json output and writes it to the given file
+  -seed int
+    	provides the seed for randomization, defaults to current unix epoch (default 1553189402)
+```
+
+#### Configuration File
+
+The gengen configuration file defines the artifacts produced.
+
+- `keys` defines the number of keys which will be produced
+- `preAlloc` is an array defining the amount of FIL for the key
+- `miners` is an array defining miners, the `owner` is the key index, and `powe` is the amount of power the miner will have in the genesis block.
+
+Example
+
+```json
+{
+  "keys": 5,
+  "preAlloc": [
+    "1000000000000",
+    "1000000000000",
+    "1000000000000",
+    "1000000000000",
+    "1000000000000"
+  ],
+  "miners": [{
+    "owner": 0,
+    "power": 1
+  }]
+}
+```
```

Comment:
I'm not too familiar with the configuration (and it honestly needs to be updated as it's a bit overly complex to describe). Suggestions are welcome, but you'll need to write out exact wording.

Reply:
I think "owner" is an index into the keys, so you could clarify that above.

Reply:
It's already stated. I'm happy to update, but you'll need to provide exact wording as I said above, the configuration is overly complex to describe. I'd have to write in paragraph form to describe how it actually works.

> the `owner` is the key index